### PR TITLE
Fix logger setup

### PR DESCRIPTION
**Pull request by barryyosi-panw**
## Related Issues
fixes: [CIAC-12159](https://jira-dc.paloaltonetworks.com/browse/CIAC-12159)

## Description

When using 'demisto-sdk' as a library within content scripts/integrations, the new logging setup/initialization conflicts with existing logger and causes the container to crash when initialized statically.

Additionally, logs does not show up on default python's logging which is used in content scripts/integration.

## Solution:
 - Excluding sdk's logging initialization logic when imported as a library.
 - Propagating log entries to default logging system when flagged.

### CODE REVIEW

File: demisto_sdk/commands/common/logger.py [hunk 1]
```diff
@@ -93,39 +98,63 @@ def logging_setup(
...
+    - path (Union[Path, str], optional): Path for file logs. If None, defaults to the calculated log directory.
+    - initial (bool): Indicates if the setup is for the initial configuration (console-only if True).
+    - propagate (bool): If True, propagates logs to Python's logging system.
+
     """
     global logger
     _setup_neo4j_logger()
 
-    logger.remove(None)  # Removes all pre-existing handlers
+    logger.remove()  # Removes all pre-existing handlers
```
Comment (dorschw) [inline demisto_sdk/commands/common/logger.py:119]: you sure passing None vs nothing still achieves what we wanted?

File: .changelog/4657.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Fixing logger crashes when imported as library in content scripts/integrations.
```
Comment (dorschw) [inline .changelog/4657.yml:2]:
```suggestion
- description: Fixing logger crashes when imported as library.
```

**Review by dorschw — COMMENTED** [review]
nice

File: demisto_sdk/commands/common/logger.py (hunk 1 above)
Comment (barryyosi-panw) [reply demisto_sdk/commands/common/logger.py:119]: Default value is None than yes, and it is less confusing that way imo. https://loguru.readthedocs.io/en/stable/api/logger.html#loguru._logger.Logger.remove

**Review by thefrieddan1 — APPROVED** [review]

**Pull request merged**